### PR TITLE
Allowing data loading after selecting/deselecting options - Long-term data > Query Log

### LINKED ISSUE #1750: "Query status" check boxes in "Long-term data" --> "Query Log" don't have any effect
### Versions

Pi-hole version is v5.2.4 (Latest: v5.2.4)
AdminLTE version is v5.4 (Latest: v5.4)
FTL version is v5.7 (Latest: v5.7)

### Platform

- OS and version:  Raspbian GNU/Linux 10 (buster)

- Platform:  Raspberry Pi

### Expected behavior

Unchecking the "Query status" boxes updates the results in the "Recent Queries" section.

### Actual behavior / bug

Checking or unchecking has no effect at all.
I don't see any button to refresh the results after (un)checking the boxes.

### Steps to reproduce

1. Go to <IP>/admin/db_queries.php (Long-term data --> Query Log)
2. Check and uncheck any check box, e.g. Permitted: forwarded

## Debug Token

- URL:  https://tricorder.pi-hole.net/qsn37j2h0r

### Screenshots

image

### PR DESCRIPTION
**Pull request by rdwebdesign**
**By submitting this pull request, I confirm the following:** 

- [x] I have read and understood the [contributors guide](https://github.com/pi-hole/AdminLTE/blob/master/CONTRIBUTING.md), as well as this entire template.
- [x] I have made only one major change in my proposed changes.
- [ ] I have commented my proposed changes within the code.
- [x] I have tested my proposed changes.
- [x] I am willing to help maintain this change if there are issues with it later.
- [x] I give this submission freely and claim no ownership.
- [x] It is compatible with the [EUPL 1.2 license](https://opensource.org/licenses/EUPL-1.1)
- [x] I have squashed any insignificant commits. ([`git rebase`](http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html))
- [x] I have Signed Off all commits. ([`git commit --signoff`](https://git-scm.com/docs/git-commit#Documentation/git-commit.txt---signoff))

---

**What does this PR aim to accomplish?:**

Fixes #1750 

**How does this PR accomplish the above?:**

On page: Long-term data  - Query Log, usually people first select the datarange field and only after that they modify the checkboxes.

This causes confusion as the data is loaded right after the date is selected. Nothing changes when clicking the checkboxes. 

This PR adds a notice and a button to allow reload the data after any checkbox changes its value.

**What documentation changes (if any) are needed to support this PR?:**

I'm not sure, but I think none.

### CODE REVIEW

**Comment by yubiuser** [comment]
Thanks for fixing so many issues lately. I'll look into this PR tmw.

**Comment by rdwebdesign** [comment]
> Thanks for fixing so many issues lately. I'll look into this PR tmw.

I never have a lot of time to do this, but recently some free time has come up.

I took the opportunity to better configure my pi-hole system. I liked the admin system, but I noticed some small bugs and details that I'd like to fix and I ended up finding the issues page on github.

I didn't expect to make so many contributions, but here I am.

**Comment by DL6ER** [comment]
> I never have a lot of time to do this, but recently some free time has come up.

> I didn't expect to make so many contributions, but here I am.

This is exactly how I started some five years ago :smile:

**Comment by yubiuser** [comment]
Bildschirmfoto zu 2021-11-06 12-09-37

This is a good solution to the issue.

Two minor things: 

1) I would make the note a bit more prominent or use this function
https://github.com/pi-hole/AdminLTE/blob/ee50954be76fa53bdc4d95a3dd197030b96af5ab/scripts/pi-hole/js/utils.js#L103

to warn the user.

2). The reload button and the note already appear even if no time range has been selected yet (hence no data fetched). This happens if you enter the site and deselect options. Can you suppress the note and button if no data has been loaded yet?

**Comment by rdwebdesign** [comment]
@yubiuser 

> 1. I would make the note a bit more prominent or use this function

I will try a different format for this, with different style. Maybe an alert... I will test.
<br>

> 2). The reload button and the note already appear even if no time range has been selected yet (hence no data fetched). This happens if you enter the side and deselect options.

Hmmm... I didn't notice that.
But it works even with no data fetched. There is a default range:
https://github.com/pi-hole/AdminLTE/blob/master/scripts/pi-hole/js/db_queries.js#L10-L13
<br>
 
> Can you suppress the note and button if no data has been loaded yet?

I will look into it, but I need to find a way to identify that the user hasn't selected any date ranges yet.

**Comment by yubiuser** [comment]
You might be able to use `showCalendar.daterangepicker` or better `apply.daterangepicker` to set a variable `PickerWasUsedByHuman=true`

from https://www.daterangepicker.com/#events

**Comment by rdwebdesign** [comment]
> You might be able to use `showCalendar.daterangepicker` or better `apply.daterangepicker` to set a variable `PickerWasUsedByHuman=true`

I will use "apply". It is already used to fire the table refresh.

**Comment by rdwebdesign** [comment]
Since we are fixing this page, what do you think of this modification?
db_queries

**Comment by yubiuser** [comment]
Looks better.

**Comment by rdwebdesign** [comment]
@PromoFaux 

you changed the number of days subtracted [here](https://github.com/pi-hole/AdminLTE/blob/master/scripts/pi-hole/js/db_queries.js#L53-L54).

I think this should be changed on **line 10**, too.

What do you think?

**Comment by PromoFaux** [comment]
Wasn't me :) but yes, I agree

**Comment by yubiuser** [comment]
Ready for review?

File: db_queries.php [hunk 1]
```diff
@@ -22,6 +22,12 @@
         </h3>
       </div>
       <div class="box-body">
+        <div class="alert alert-info reload-box">
+            <div>
+                <span><i class="fa fa-exclamation-circle"></i>&nbsp; New options selected. Please, reload the data or choose a another time range.</span>
```
Comment (yubiuser) [inline db_queries.php]:
```suggestion
                <span><i class="fa fa-exclamation-circle"></i>&nbsp; New options selected. Please reload the data or choose another time range.</span>
```

**Comment by yubiuser** [comment]
Could you please also fix the 7 days also here

https://github.com/pi-hole/AdminLTE/blob/ee50954be76fa53bdc4d95a3dd197030b96af5ab/scripts/pi-hole/js/db_graph.js#L10

and

https://github.com/pi-hole/AdminLTE/blob/ee50954be76fa53bdc4d95a3dd197030b96af5ab/scripts/pi-hole/js/db_lists.js#L10

**Comment by rdwebdesign** [comment]
I tried to commit directly on github web interface and I got this error. Can You revert it?

**Comment by yubiuser** [comment]
It was only DCO not to pass. You signed all your previous commits, so I set it to pass.

**Review by yubiuser — DISMISSED** [review]

**Comment by yubiuser** [comment]
I was to fast... the 7 days fix is not included.

**Comment by rdwebdesign** [comment]
No it's not.

I'm starting the changes now.

**Comment by rdwebdesign** [comment]
This time I did not use the web interface and I signed off, but DCO is complaining again. 
Please, can you set it to pass?

**Comment by DL6ER** [comment]
Thanks for your work. Just FYI: this page is going to be removed in the upcoming Pi-hole v6.0 (the source code is here on Github and needs to get `devel` merged into it) as we plan to provide a unified Query Log that allows advanced filtering, the implementation is pretty rough on the style/design side of things but actually works (which is more important):

192 168 2 11_8080_admin_queries php

In the end, so much has changed that we may need to have to start doing the v6.0 transition work again and will then use what you added herein.

**Comment by yubiuser** [comment]
> This time I did not use the web interface and I signed off, but DCO is complaining again.
Please, can you set it to pass?

Sure. If only one commit in the whole PR is not singned-off DCO will repeatedly complain.

**Review by yubiuser — APPROVED** [review]

**Comment by rdwebdesign** [comment]
@DL6ER 

> Thanks for your work. Just FYI: this page is going to be removed in the upcoming Pi-hole v6.0 (the source code is here on Github and needs to get `devel` merged into it) as we plan to provide a unified Query Log that allows advanced filtering, the implementation is pretty rough on the style/design side of things but actually works (which is more important):

I don't know which branch to look for... Where is the v6.0 source code?

**Comment by DL6ER** [comment]
https://github.com/pi-hole/AdminLTE/tree/new/FTL_is_my_new_home

Arguably not the best branch name but this goes back quite some time. As said above, will likely have to start from scratch as it will be easier than two-way merging. But that this also needs FTL from the v6.0 branch (`new/http`) for the REST API and the embedded webserver.

**Comment by rdwebdesign** [comment]
Ok... it's a lot of different new parts mixed together.
I leave to see this later, when I better understand the code.

**Pull request merged**